microservices/product-search/Dockerfile

```diff
@@ -0,0 +1,36 @@
+FROM php:7.2-fpm-alpine
+
+# install Composer
+COPY docker/docker-install-composer /usr/local/bin/docker-install-composer
+RUN chmod +x /usr/local/bin/docker-install-composer && docker-install-composer
+
+# install necessary PHP extensions
+RUN docker-php-ext-install \
+    opcache
+
+# install locales and switch to en_US.utf8 in order to enable UTF-8 support
+# see https://github.com/docker-library/php/issues/240#issuecomment-305038173
+RUN apk add --update --no-cache --repository http://dl-cdn.alpinelinux.org/alpine/edge/testing gnu-libiconv
+ENV LD_PRELOAD /usr/lib/preloadable_libiconv.so php
+ENV LC_ALL=en_US.utf8 LANG=en_US.utf8 LANGUAGE=en_US.utf8
+
+# overwrite the original entry-point from the PHP Docker image with our own
+COPY docker/docker-php-entrypoint /usr/local/bin/
+
+# copy source code to container
+COPY / /var/www/html
+
+ARG github_token
+RUN composer config -g github-oauth.github.com $github_token
+RUN composer install
+
+# allow overwriting UID and GID o the user "www-data" to help solve issues with permissions in mounted volumes
+# if the GID is already in use, we will assign GID 82 instead (82 is the standard uid/gid for "www-data" in Alpine)
+ARG www_data_uid
+ARG www_data_gid
+RUN deluser www-data && (addgroup -g $www_data_gid www-data || addgroup -g 82 www-data) && adduser -u $www_data_uid -D -S -G www-data www-data
```

Comment:
Is this whole part with www-data necessary for deployment? Because www-data and their IDs were implemented because of local development. But I think that this is not necessary in this

Reply:
This Dockerfile is used for building in the monorepo so you need to set up permissions here.

Reply:
I think that is not a problem plus you can anytime choose to create your own Dockerfile using this one and permission is solved for you already.